scripts/data_prep/convert_delta_to_json.py

```diff
@@ -0,0 +1,451 @@
+        collect (bool): whether to get the underlying data from spark dataframe
+    """
+    if method not in ['dbsql', 'dbconnect']:
+        raise ValueError(f'Unrecognized method: {method}')
+
+    if method == 'dbsql':
+        if cursor is None:
+            raise ValueError(f'cursor cannot be None if using method dbsql')
+        cursor.execute(q)
+        if collect:
+            return cursor.fetchall()
+
+    elif method == 'dbconnect':
+        if spark == None:
+            raise ValueError(f'sparkSession is required for dbconnect')
+        df = spark.sql(q)
+        if collect:
+            return df.collect()
+        return df
+
+    return None
+
+
+def get_args(signed: List, json_output_path: str, columns: List,
+             cf_collect_type: str) -> Iterable:
+    for i, r in enumerate(signed):
+        yield (i, r.url, json_output_path, columns, cf_collect_type)
+
+
+def download(ipart: int,
+             url: str,
+             json_output_path: str,
+             columns: Optional[List] = None,
+             resp_format: str = 'arrow',
+             compressed: bool = False) -> None:
+    """Thread download presigned url and save to jsonl locally.
+
+    Args:
+        ipart (int): presigned url id
+        url (str): presigned url
+        json_output_path (str): directory to save the ipart_th segment of dataframe
+        columns (list): schema to save to json
+        resp_format (str): whether to use arrow or json when collect
+        compressed (bool): if data is compressed before downloading. Need decompress if compressed=True.
+    """
+    resp = requests.get(url)
+    if resp.status_code == 200:
+        if resp_format == 'json':
+            data = resp.json()
+            pd.DataFrame(data, columns=columns).to_json(os.path.join(
+                json_output_path, 'part_' + str(ipart) + '.jsonl'),
+                                                        orient='records',
+                                                        lines=True)
+            return
+
+        if compressed:
+            # The data is lz4 compressed arrow format.
+            # Decompress the data
+            decompressed_data = lz4.frame.decompress(resp.content)
+            # Convert the decompressed data into a PyArrow table
+            reader = pa.ipc.open_stream(decompressed_data)
+        else:
+            reader = pa.ipc.open_stream(resp.content)
+        table = reader.read_all()
+
+        # Convert the PyArrow table into a pandas DataFrame
+        df = table.to_pandas()
+        df.to_json(os.path.join(json_output_path,
+                                'part_' + str(ipart) + '.jsonl'),
+                   orient='records',
+                   lines=True,
+                   force_ascii=False)
+
+
+def download_starargs(args: Tuple) -> None:
+    return download(*args)
+
+
+def fetch_data(method: str, cursor: Optional[Cursor],
+               sparkSession: Optional[SparkSession], s: int, e: int,
```

Comment:
better name than `s` and `e`. Maybe `start` and `end`?

Also method comment is missing.